addons/controllers/vspherecpiconfig_controller_test.go

```diff
@@ -250,6 +253,48 @@ var _ = Describe("VSphereCPIConfig Reconciler", func() {
 		})
 	})
 
+	Context("reconcile VSphereCPIConfig manifests in non-paravirtual mode, with multi-tenancy enabled", func() {
+
+		identity := &capvv1beta1.VSphereClusterIdentity{}
+		identitySecret := &v1.Secret{}
+		identityNamespace := "capv-system"
+
+		BeforeEach(func() {
+			clusterName = "test-cluster-cpi-multi-tenancy"
+			clusterResourceFilePath = "testdata/test-vsphere-cpi-non-paravirtual-multi-tenancy.yaml"
+		})
+
+		JustAfterEach(func() {
+			Expect(testutil.DeleteNamespace(ctx, clientSet, identityNamespace)).To(Succeed())
+		})
+
+		It("Should reconcile VSphereCPIConfig and create data values secret for VSphereCPIConfig on management cluster", func() {
+			identityName := "multi-tenancy"
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Name: identityName}, identity)).To(Succeed())
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Name: identityName, Namespace: identityNamespace}, identitySecret)).To(Succeed())
+
+			identity.Status.Ready = true
+			Expect(k8sClient.Status().Update(ctx, identity)).To(Succeed())
+
+			secret := &v1.Secret{}
+			Eventually(func() bool {
+				secretKey := client.ObjectKey{
+					Namespace: clusterNamespace,
+					Name:      fmt.Sprintf("%s-%s-data-values", clusterName, constants.CPIAddonName),
+				}
+				if err := k8sClient.Get(ctx, secretKey, secret); err != nil {
+					return false
+				}
+				secretData := string(secret.Data["values.yaml"])
+				Expect(len(secretData)).ShouldNot(BeZero())
+				Expect(strings.Contains(secretData, "username: administrator@vsphere.local")).Should(BeTrue())
+				Expect(strings.Contains(secretData, "password: Admin!23")).Should(BeTrue())
```

Comment:
Maybe better not to commit these credentials here?

Reply:
thanks, just fixed